layer/components/Lego/GithubStar/index.vue

```diff
@@ -0,0 +1,38 @@
+<script lang="ts" setup>
+import { withBase, withoutBase } from 'ufo'
+import { useFetch } from '#imports'
+
+const props = defineProps<{
+  repo: string
+  raw?: boolean
+  to?: string
+}>()
+
+const repo = computed(() => {
+  // support users providing full github url
+  return withoutBase(props.repo, 'https://github.com/')
+})
+const link = computed(() => {
+  return props.to || withBase(repo.value, 'https://github.com/')
+})
+// pull the stars from the server
+const { data } = await useFetch('/api/get-github-stars', {
+  query: {
+    repo,
+  },
+})
```

Comment:
Wonder why this trigger a refetch when client loaded, and `Tweet/index.vue` doesn't. 

Maybe we can fetch from `ungh` directly? Then it would work with SSG.

(This website is using SSG, thus showing NaN)
https://github.com/zernonia/nuxt-lego/assets/59365435/975558e9-bce5-4da1-96e2-89dc45d36540

Reply:
Good point, I believe the issue was providing a ref as a query. I'm guessing Nuxt messes up the key generation, will look into that separately.

New logic seems to fix it.